official/utils/misc/distribution_utils.py

```diff
@@ -96,3 +96,51 @@ def per_device_batch_size(batch_size, num_gpus):
           ).format(num_gpus, batch_size, batch_size - remainder)
     raise ValueError(err)
   return int(batch_size / num_gpus)
+
+
```

Comment:
Add a big comment here saying this is a temporary measure and will be removed when we have better support for this type of synthetic data maybe in tf.data or distribution strategies.